types/src/transaction/mod.rs

```diff
@@ -908,6 +910,37 @@ impl TransactionOutput {
     }
 }
 
+pub trait TransactionInfoTrait:
+    Clone + CryptoHash + Debug + Display + Eq + Serialize + for<'de> Deserialize<'de>
+{
+    /// Constructs a new `TransactionInfo` object using transaction hash, state root hash and event
+    /// root hash.
+    fn new(
+        transaction_hash: HashValue,
+        state_root_hash: HashValue,
+        event_root_hash: HashValue,
+        gas_used: u64,
+        status: KeptVMStatus,
+    ) -> Self;
+
+    /// Returns the hash of this transaction.
+    fn transaction_hash(&self) -> HashValue;
+
+    /// Returns root hash of Sparse Merkle Tree describing the world state at the end of this
+    /// transaction.
+    fn state_root_hash(&self) -> HashValue;
+
+    /// Returns the root hash of Merkle Accumulator storing all events emitted during this
+    /// transaction.
+    fn event_root_hash(&self) -> HashValue;
+
+    /// Returns the amount of gas used by this transaction.
+    fn gas_used(&self) -> u64;
+
+    /// Resturns the Status from the VM for this transaction.
+    fn status(&self) -> &KeptVMStatus;
+}
```

Comment:
Do you have an example of what the shape of the new transaction info type you want to use will look like?

Reply:
The  new `TransactionInfo` (probably a variant of an enum this time?) will have the state root optional, and a root hash summarizing all the state changes local to a transaction -- the trait interface needs to change later on, this is here to only demo the possibility of parameterize on these things (and these things grouped under a cover-all ProtocolSpec trait).

Reply:
Given this is apart of the public interface of the blockchain, how are you envisioning this transition will happen? Breaking change?

Reply:
I think in the current json-rpc client as the main interface world, it's not necessarily breaking, as in people talk in the json-rpc "xView" types, not the native types. But anyway this will happen soonish (in a few weeks). The plan is to stablize the data traits (as well as the core logic connecting them and core benchmarks) before migrating the entire toolchain (the node, the tools, the B&R system, etc)